CMakeLists.txt

```diff
@@ -410,61 +431,61 @@ hpx_option(HPX_WITH_ITTNOTIFY BOOL "Enable Amplifier (ITT) instrumentation suppo
+hpx_option(HPX_WITH_VERIFY_LOCKS_GLOBALLY BOOL
   "Enable global lock verification code (default: OFF, implicitly enabled in debug builds)"
   OFF
   CATEGORY "Debugging" ADVANCED)
-hpx_option(HPX_VERIFY_LOCKS_BACKTRACE BOOL
-  "Enable thred stack back trace being captured on lock registration (to be used in combination with HPX_VERIFY_LOCKS=ON, default: OFF)"
+hpx_option(HPX_WITH_VERIFY_LOCKS_BACKTRACE BOOL
+  "Enable thred stack back trace being captured on lock registration (to be used in combination with HPX_WITH_VERIFY_LOCKS=ON, default: OFF)"
   OFF
   CATEGORY "Debugging" ADVANCED)
-hpx_option(HPX_THREAD_DEBUG_INFO BOOL
+hpx_option(HPX_WITH_THREAD_DEBUG_INFO BOOL
   "Enable thread debugging information (default: OFF, implicitly enabled in debug builds)"
   OFF
   CATEGORY "Debugging" ADVANCED)
-hpx_option(HPX_THREAD_GUARD_PAGE BOOL
+hpx_option(HPX_WITH_THREAD_GUARD_PAGE BOOL
   "Enable thread guard page (default: ON)"
   ON
   CATEGORY "Debugging" ADVANCED)
 
-if(HPX_VERIFY_LOCKS)
+if(HPX_WITH_VERIFY_LOCKS)
   hpx_add_config_define(HPX_HAVE_VERIFY_LOCKS)
-  if(HPX_VERIFY_LOCKS_BACKTRACE)
+  if(HPX_WITH_VERIFY_LOCKS_BACKTRACE)
     hpx_add_config_define(HPX_HAVE_VERIFY_LOCKS_BACKTRACE)
   endif()
 endif()
-if(HPX_VERIFY_LOCKS_GLOBALLY)
+if(HPX_WITH_VERIFY_LOCKS_GLOBALLY)
   hpx_add_config_define(HPX_HAVE_VERIFY_LOCKS_GLOBALLY)
 endif()
 
 # Additional debug support
-if(NOT MSVC AND HPX_THREAD_GUARD_PAGE)
-  hpx_add_config_define(HPX_THREAD_GUARD_PAGE)
+if(NOT MSVC AND HPX_WITH_THREAD_GUARD_PAGE)
+  hpx_add_config_define(HPX_HAVE_THREAD_GUARD_PAGE)
 endif()
 
-if(NOT MSVC AND HPX_THREAD_STACK_MMAP)
-  hpx_add_config_define(HPX_USE_MMAP)
+if(NOT MSVC AND HPX_WITH_THREAD_STACK_MMAP)
+  hpx_add_config_define(HPX_HAVE_THREAD_STACK_MMAP)
 endif()
 
-if(HPX_THREAD_MANAGER_IDLE_BACKOFF)
-  hpx_add_config_define(HPX_THREAD_BACKOFF_ON_IDLE)
+if(HPX_WITH_THREAD_MANAGER_IDLE_BACKOFF)
+  hpx_add_config_define(HPX_HAVE_THREAD_MANAGER_IDLE_BACKOFF)
 endif()
 
-if(HPX_THREAD_DEBUG_INFO)
-  hpx_add_config_define(HPX_THREAD_MAINTAIN_PARENT_REFERENCE)
-  hpx_add_config_define(HPX_THREAD_MAINTAIN_PHASE_INFORMATION)
-  hpx_add_config_define(HPX_THREAD_MAINTAIN_DESCRIPTION)
-  hpx_add_config_define(HPX_THREAD_MAINTAIN_DEADLOCK_DETECTION)
+if(HPX_WITH_THREAD_DEBUG_INFO)
+  hpx_add_config_define(HPX_HAVE_THREAD_PARENT_REFERENCE)
+  hpx_add_config_define(HPX_HAVE_THREAD_PHASE_INFORMATION)
+  hpx_add_config_define(HPX_HAVE_THREAD_DESCRIPTION)
+  hpx_add_config_define(HPX_HAVE_THREAD_DEADLOCK_DETECTION)
 endif()
 
 # run hpx_main on all localities by default
-hpx_option(HPX_RUN_MAIN_EVERYWHERE BOOL "Run hpx_main by default on all localities (default: OFF)." OFF ADVANCED)
-if(HPX_RUN_MAIN_EVERYWHERE)
-  hpx_add_config_define(HPX_RUN_MAIN_EVERYWHERE)
+hpx_option(HPX_WITH_RUN_MAIN_EVERYWHERE BOOL "Run hpx_main by default on all localities (default: OFF)." OFF ADVANCED)
+if(HPX_WITH_RUN_MAIN_EVERYWHERE)
+  hpx_add_config_define(HPX_HAVE_RUN_MAIN_EVERYWHERE)
```

Comment:
I know that this has been there since a while, but I think it's harmful to have that set globally. It would make more sense to have that option on a application by application basis.